airflow/providers/google/cloud/sensors/gcs.py

```diff
@@ -514,3 +518,40 @@ def poke(self, context: Context) -> bool:
         return self.is_bucket_updated(
             set(self._get_gcs_hook().list(self.bucket, prefix=self.prefix))  # type: ignore[union-attr]
         )
+
+    def execute(self, context: Context) -> None:
+        """Airflow runs this method on the worker and defers using the trigger."""
+        hook_params = {"impersonation_chain": self.impersonation_chain}
+
+        if not self.deferrable:
+            super().execute(context)
+            return
```

Comment:
```suggestion
            return super().execute(context)
```

Otherwise any possible return value won't make it to xcom

Reply:
Fixed in [3e42e43](https://github.com/apache/airflow/pull/31081/commits/3e42e43c93597d5e5d1a25eb488521725fc9b561)